pahfit/features/features.py

```diff
@@ -308,16 +308,16 @@ def _construct_table(cls, inp: dict):
         tables = []
         for (kind, features) in inp.items():
             kind_params = cls._kind_params[kind]  # All params for this kind
-            rows = []
+            rows, dtypes = [], []
             for (name, params) in features.items():
                 for missing in kind_params - params.keys():
                     if missing in cls._no_bounds:
                         params[missing] = 0.0
                     else:
                         params[missing] = value_bounds(0.0, bounds=(0.0, None))
                 rows.append(dict(name=name, **params))
-            table_columns = rows[0].keys()
-            t = cls(rows, names=table_columns)
+                dtypes.append(None if name in cls._no_bounds else cls._bounds_dtypes)
+            t = cls(rows, names=rows[0].keys(), dtype=dtypes)
             for p in cls._kind_params[kind]:
                 if p not in cls._no_bounds:
                     t[p].info.format = "0.4g"  # Nice format (customized by Formatter)
```

Comment:
With my changes explaiend above applied, we're getting somewhere. But `Features.read(packfile)` is crashing on this line. When "0.4g" is assigned to t[p].info.format, the astropy code will perform a test of the assigned formatting function. This ultimately results in the following errors (multiple, because some are caught after which another is raised)
```
E               ValueError: unable to parse format string 0.4g for its column.
E                   ValueError: Unable to parse format string "0.4g" for entry "(0., 0., inf)" in column "tau"
E           ValueError: Invalid format for column 'tau': could not display values in this column using this format
```

Reply:
I removed this line (replaced by `pass`) to see what would happen. The code now continues, and here's a quick test
```python
f = Features.read('classic.yaml')
# change one of the bounds. I like this more explicit syntax.
f.loc['PAH_5.3']['fwhm']['max'] = 0.3 
f.pprint()
```

The output is 
```
In [8]: f.pprint()
    name       group          kind      temperature [val, min, max] ... fwhm [val, min, max]      model      geometry
                                                     K              ...          um                                  
----------- ------------ -------------- --------------------------- ... -------------------- --------------- --------
  starlight       _none_      starlight                5000 (fixed) ...                   --              --       --
dust_cont00    dust_cont dust_continuum                 300 (fixed) ...                   --              --       --
dust_cont01    dust_cont dust_continuum                 200 (fixed) ...                   --              --       --
dust_cont02    dust_cont dust_continuum                 135 (fixed) ...                   --              --       --
dust_cont03    dust_cont dust_continuum                  90 (fixed) ...                   --              --       --
dust_cont04    dust_cont dust_continuum                  65 (fixed) ...                   --              --       --
dust_cont05    dust_cont dust_continuum                  50 (fixed) ...                   --              --       --
dust_cont06    dust_cont dust_continuum                  40 (fixed) ...                   --              --       --
dust_cont07    dust_cont dust_continuum                  35 (fixed) ...                   --              --       --
    H2_S(7)     H2_lines           line                          -- ...                   --              --       --
    H2_S(6)     H2_lines           line                          -- ...                   --              --       --
    H2_S(5)     H2_lines           line                          -- ...                   --              --       --
    H2_S(4)     H2_lines           line                          -- ...                   --              --       --
    H2_S(3)     H2_lines           line                          -- ...                   --              --       --
    H2_S(2)     H2_lines           line                          -- ...                   --              --       --
    H2_S(1)     H2_lines           line                          -- ...                   --              --       --
    H2_S(0)     H2_lines           line                          -- ...                   --              --       --
     [ArII]  ionic_lines           line                          -- ...                   --              --       --
    [ArIII]  ionic_lines           line                          -- ...                   --              --       --
      [SIV]  ionic_lines           line                          -- ...                   --              --       --
     [NeII]  ionic_lines           line                          -- ...                   --              --       --
    [NeIII]  ionic_lines           line                          -- ...                   --              --       --
  [SIII]_18  ionic_lines           line                          -- ...                   --              --       --
      [OIV]  ionic_lines           line                          -- ...                   --              --       --
  [FeII]_26  ionic_lines           line                          -- ...                   --              --       --
  [SIII]_33  ionic_lines           line                          -- ...                   --              --       --
     [SiII]  ionic_lines           line                          -- ...                   --              --       --
  [FeII]_35  ionic_lines           line                          -- ...                   --              --       --
    PAH_5.3       _none_   dust_feature                          -- ...    0.17918 (-∞, 0.3)              --       --
    PAH_5.7       _none_   dust_feature                          -- ...       0.1995 (fixed)              --       --
    PAH_6.2       _none_   dust_feature                          -- ...       0.1866 (fixed)              --       --
    PAH_6.7       _none_   dust_feature                          -- ...       0.4683 (fixed)              --       --
   PAH_7.7a  PAH_7.7_cmp   dust_feature                          -- ...      0.93492 (fixed)              --       --
   PAH_7.7b  PAH_7.7_cmp   dust_feature                          -- ...       0.3344 (fixed)              --       --
   PAH_7.7c  PAH_7.7_cmp   dust_feature                          -- ...      0.41605 (fixed)              --       --
    PAH_8.3       _none_   dust_feature                          -- ...       0.4165 (fixed)              --       --
    PAH_8.6       _none_   dust_feature                          -- ...      0.33579 (fixed)              --       --
   PAH_10.7       _none_   dust_feature                          -- ...       0.2136 (fixed)              --       --
  PAH_11.3a PAH_11.3_cmp   dust_feature                          -- ...      0.13476 (fixed)              --       --
  PAH_11.3b PAH_11.3_cmp   dust_feature                          -- ...      0.36256 (fixed)              --       --
     PAH_12       _none_   dust_feature                          -- ...      0.53955 (fixed)              --       --
  PAH_12.6a PAH_12.6_cmp   dust_feature                          -- ...      0.53004 (fixed)              --       --
  PAH_12.6b PAH_12.6_cmp   dust_feature                          -- ...      0.16497 (fixed)              --       --
  PAH_13.48       _none_   dust_feature                          -- ...       0.5392 (fixed)              --       --
  PAH_14.04       _none_   dust_feature                          -- ...      0.22464 (fixed)              --       --
  PAH_14.19       _none_   dust_feature                          -- ...      0.35475 (fixed)              --       --
   PAH_15.9       _none_   dust_feature                          -- ...        0.318 (fixed)              --       --
    PAH_17a   PAH_17_cmp   dust_feature                          -- ...       0.2303 (fixed)              --       --
    PAH_17b   PAH_17_cmp   dust_feature                          -- ...       1.1076 (fixed)              --       --
    PAH_17c   PAH_17_cmp   dust_feature                          -- ...       0.2085 (fixed)              --       --
    PAH_17d   PAH_17_cmp   dust_feature                          -- ...      0.28592 (fixed)              --       --
   PAH_33.1       _none_   dust_feature                          -- ...        1.655 (fixed)              --       --
   silicate       _none_    attenuation                          -- ...                   -- S07_attenuation    mixed
```

Reply:
Yeah that was dumb to leave in.  All `no_bounds` are simply strings, so leave it alone and `Table` will sort that out.

Reply:
Wait never-mind.  The idea is that the custom formatter shims this column format (which the user can change) in at the last second prior to printing.  But I guess this is being defeated by this testing, which happens outside the printer.  Sigh.  I want the user to be able to easily customize `col.info.format="%5.2f"` or whatever.  And to not worry about removing the function from that slot when we save (which results in an error).

Reply:
Maybe that's too cute.  Maybe we should just set up `col.info.format = fmt_func` right from the get-go, then fix up `Features.write` to replace it "just in time".  If you leave a function in that slot, you get:

```python
RepresenterError: ('cannot represent an object', <function _fmt at 0x118bedbc0>)
```

We could tell the user to alter `col.info.format2` or something if we want to use that in our formatter.  Thoughts?

Reply:
Having an extra attribute for the "individual number" format seems like the most reliable way. Our _pformat_table can then grab this value to generate the right "val min max" formatting function with fmt_func(col.info.format2). 

Do we need a more informative name as opposed to format2? Maybe something that makes it more clear that it's a modification by pahfit?

Reply:
OK I have stamped out the bugs here I think.  I just used `str` for all the `no_bounds` stuff (adding `kind`).

Reply:
Update: I do now allow `meta['pahfit_format']` on both table and columns.  Works great.